packages/toolbox-core/src/toolbox_core/sync_client.py

```diff
@@ -42,125 +40,105 @@ def __init__(
         url: str,
     ):
         """
-        Initializes the SyncToolboxClient.
+        Initializes the ToolboxClient.
 
         Args:
-            url: The base URL for the Toolbox service API (e.g., "http://localhost:8000").
+            url: The base URL for the Toolbox service API (e.g., "http://localhost:5000").
         """
         # Running a loop in a background thread allows us to support async
         # methods from non-async environments.
-        if ToolboxSyncClient.__loop is None:
+        if self.__class__.__loop is None:
             loop = asyncio.new_event_loop()
             thread = Thread(target=loop.run_forever, daemon=True)
             thread.start()
-            ToolboxSyncClient.__thread = thread
-            ToolboxSyncClient.__loop = loop
+            self.__class__.__thread = thread
+            self.__class__.__loop = loop
 
-        async def __start_session() -> None:
-            # Use a default session if none is provided. This leverages connection
-            # pooling for better performance by reusing a single session throughout
-            # the application's lifetime.
-            if ToolboxSyncClient.__session is None:
-                ToolboxSyncClient.__session = ClientSession()
+        async def create_client():
+            return ToolboxClient(url)
 
-        coro = __start_session()
+        # Ignoring type since we're already checking the existing of a loop above.
+        self.__async_client = asyncio.run_coroutine_threadsafe(
+            create_client(), ToolboxSyncClient.__loop  # type: ignore
```

Comment:
nit: To make it consistent with how it's done above?

```suggestion
            create_client(), self.__class__.__loop  # type: ignore
```